commands/configure/index.js

```diff
@@ -112,7 +129,7 @@ program
       }
       throw e;
     }
-    const config = { applicationId: appInfo.id, applicationName: appInfo.name };
+    const config = { applicationId: appInfo.id, applicationName: appInfo.name, apiUrl: url };
```

Comment:
Also, save the white label info onto the config experienceDomain and appUrl. To make this backward compatible in loadConfig function if either of these fields is missing then I would make the call to look up the white label info, I would then resave the configuration file and then return the updated object. You can do this because white label info does not change. Then you can probably remove most of those knock calls from your test.

Reply:
I like this idea, except it raises some backwards compatibility issues. What do we default to if the .application file is still in the old format? 
Since we're allowing DEFAULT_API_URL to be configurable, then the appUrl and endpointDomain need to be equally configurable, right? I can't just default them to app.losant.com and onlosant.com.

Reply:
by default, DEFAULT_API_URL will be api.losant.com right and the only way it's configurable is by overwriting the environment variable. Currently, it's only being used for your tests, but it could be used for future things...not really sure I have an exact use case at the moment. 

If you see your .application file does not have it in there when loaded using your getWhitelabel function to get the white label, resave your .application file and then return the updated config. So they either just configured the file in which case it will never try to update the .application file again and for a user who already has a configured file, this will automatically do it once.

Reply:
That's the solution I came to. Issue I have with it is that it requires nocking the whitelabel get in every test that runs loadConfig, which is (almost) all of them. Just gotta sift through those.

Reply:
couldnt you update the file that's preloaded in the tests to be this new format and then it would not require a nock?

Reply:
I thought we wanted to keep the common test file unchanged in order to properly test backwards compatibility?

Reply:
We talked about you added a test with the original credentials file that runs through configure...but no you don't have to keep it the old way, when adding new things. you will need to add a test for this function that it properly turns an original .application.yaml file into the new format but you dont need to do that for every test.